agent/lm_agent/reconciliation.py

```diff
@@ -38,91 +32,81 @@
     scontrol_update_reservation,
 )
 
-RECONCILE_URL_PATH = "/lm/api/v1/license/reconcile"
 
-
-def get_greatest_grace_time(job_id: str, grace_times: Dict[int, int], booking_rows: List) -> int:
+def get_greatest_grace_time_for_job(grace_times: Dict[int, int], job_bookings: List[BookingSchema]) -> int:
     """
-    Find the greatest grace_time for the given job_id.
+    Find the greatest grace_time for each feature booked by the given job_id.
     """
     greatest_grace_time = -1
-    for book in booking_rows:
-        if not book:
-            continue
-        for inner_book in book:
-            if str(inner_book["job_id"]) != str(job_id):
-                continue
-            config_id_for_grace_times = inner_book["config_id"]
-            greatest_grace_time = max(greatest_grace_time, grace_times[config_id_for_grace_times])
+    for booking in job_bookings:
+        feature_id = booking.feature_id
+        greatest_grace_time = max(greatest_grace_time, grace_times[feature_id])
     return greatest_grace_time
```

Comment:
```suggestion
    return max(map(lambda feature_id: grace_times[feature_id], map(lambda booking: booking.feature_id, job_bookings)))
```

Reply:
maybe generating a set with all the grace_time keys can improve as well

Reply:
I feel like using `map` here would reduce the readability.
The unit test for this function failed with `max() arg is an empty sequence` using the suggested change.
Looks like it doesn't cover the case where the job doesn't have any bookings.

Reply:
I went with @dusktreader's [suggestion](https://github.com/omnivector-solutions/license-manager/pull/227#discussion_r1237670454) for this part of the code.